README.md

```diff
@@ -19,6 +19,148 @@ It is *highly* recommended that users of the library learn [Google Protocol Buff
+* `channel_options` is an array of gRPC option tuples, such as `[("grpc.ssl_target_name_override", "ems.cisco.com")]`. This assumes some knowledge of gRPC options.
+* `from_file` indicates that the `root_certificates`, `private_key`, and `certificate_chain` variables represent file paths which should be read from the filesystem as opposed to the certificate content being passed directly in the variables.
+
+```python
+as_insecure(
+    self,
+    channel_options=None,
+    compression=None
+)
+```
+
+* `as_insecure` utilizes an insecure gRPC channel and is explicitly against gNMI specification. This option is not recommended and exists purely due to insecure server implementations existing. Utilizing `as_secure(root_from_target=True, target_name_from_root=True)` should negate most necessity or desire for this option.
+
+```python
+with_authentication(
+    self,
+    username,
+    password
+)
+```
+
+* This method simply sets the `username` and `password` to utilize for authenticating RPCs with the devices as metadata (Call credentials).
+
+#### Initialization Examples
+
+Using an encrypted channel automatically getting the certificate from the device, quick for testing:
+
+```python
+from cisco_gnmi import Client
+
+client = Client(
+    '127.0.0.1:9339',
+    attempt_implicit_secure=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+
+# More explicitly...
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    root_from_target=True,
+    target_name_from_root=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+```
+
+
+Using an owned root certificate on the filesystem:
+
+```python
+from cisco_gnmi import Client
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    'ems.pem',
+    from_file=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+```
+
+Passing certificate content to method:
+
+```python
+from cisco_gnmi import Client
+
+# Note reading as bytes
+with open('ems.pem', 'rb') as cert_fd:
+    root_cert = cert_fd.read()
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    root_cert
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+```
+
+Usage with root certificate, private key, and cert chain:
+
+```python
+from cisco_gnmi import Client
+
+client = Client(
+    '127.0.0.1:9339'
+).as_secure(
+    root_certificates='rootCA.pem',
+    private_key='client.key',
+    certificate_chain='client.crt',
+    from_file=True
+).with_authentication(
+    'admin',
+    'its_a_secret'
+)
+```
+
+
 ### XRClient
 `XRClient` inherets from `Client` and provides several wrapper methods which aid with IOS XR-specific behaviors of the gNMI implementation. These are `delete_xpaths`, `get_xpaths`, `set_json`, and `subscribe_xpaths`. These methods make several assumptions about what kind of information will be supplied to them in order to simplify usage of the gNMI RPCs.
```

Comment:
Do you want to specify what assumptions are made?

Reply:
Not necessarily, the method doc and parameters themselves kind of embody what those assumptions are.

Reply:
Called out the assumptions being in the doc.